cmd/account/generate-secret.go

```diff
@@ -0,0 +1,319 @@
+
+	"github.com/openshift/osd-utils-cli/cmd/common"
+	"github.com/openshift/osd-utils-cli/pkg/k8s"
+	awsprovider "github.com/openshift/osd-utils-cli/pkg/provider/aws"
+)
+
+// newCmdGenerateSecret implements the generate-secret command which generates an new set of IAM User credentials
+func newCmdGenerateSecret(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	ops := newGenerateSecretOptions(streams, flags)
+	generateSecretCmd := &cobra.Command{
+		Use:               "generate-secret <IAM User name>",
+		Short:             "Generate IAM credentials secret",
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+		Aliases: []string{"generate-secrets"},
+	}
+
+	generateSecretCmd.Flags().StringVarP(&ops.accountName, "account-name", "a", "", "AWS Account CR name")
+	generateSecretCmd.Flags().StringVar(&ops.accountNamespace, "account-namespace", common.AWSAccountNamespace,
+		"The namespace to keep AWS accounts. The default value is aws-account-operator.")
+	generateSecretCmd.Flags().StringVarP(&ops.accountID, "account-id", "i", "", "AWS Account ID")
+	generateSecretCmd.Flags().StringVarP(&ops.profile, "aws-profile", "p", "", "specify AWS profile")
+	generateSecretCmd.Flags().StringVar(&ops.secretName, "secret-name", "", "Specify name of the generated secret")
+	generateSecretCmd.Flags().StringVar(&ops.secretNamespace, "secret-namespace", "aws-account-operator", "Specify namespace of the generated secret")
+	generateSecretCmd.Flags().BoolVar(&ops.quiet, "quiet", false, "Suppress logged output")
+	generateSecretCmd.Flags().BoolVar(&ops.ccs, "ccs", false, "Only generate specific secret for osdCcsAdmin. Requires Account CR name")
+
+	return generateSecretCmd
+}
+
+// generateSecretOptions defines the struct for running generate-secret command
+type generateSecretOptions struct {
+	accountName      string
+	accountID        string
+	accountNamespace string
+	iamUsername      string
+
+	secretName      string
+	secretNamespace string
+	quiet           bool
+	ccs             bool
+	outputPath      string
+
+	// AWS config
+	region  string
+	profile string
+	cfgFile string
+
+	flags *genericclioptions.ConfigFlags
+	genericclioptions.IOStreams
+	kubeCli client.Client
+}
+
+func newGenerateSecretOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *generateSecretOptions {
+	return &generateSecretOptions{
+		flags:     flags,
+		IOStreams: streams,
+	}
+}
+
+func (o *generateSecretOptions) complete(cmd *cobra.Command, args []string) error {
+
+	// Opinionated config for CCS rotation
+	if o.ccs {
+		// If no account CR name is provided via the -a option, try and get it via argument
+		if o.accountName == "" {
+			if len(args) != 1 {
+				return cmdutil.UsageErrorf(cmd, "Account CR name argument is required")
+			}
+			o.accountName = args[0]
```

Comment:
I don't get why we populate `o.accountName` from `args[0]` here, and below on L101 we populate `o.iamUsername` from the same arg.

Shouldn't only the IAM Username be passed as a cli arg, while accountName should come via `-a account-name`?